sql/hive/src/test/scala/org/apache/spark/sql/hive/MetastoreDataSourcesSuite.scala

```diff
@@ -1403,6 +1403,69 @@ class MetastoreDataSourcesSuite extends QueryTest with SQLTestUtils with TestHiv
     }
   }
 
+  test("SPARK-37283: Don't try to store a V1 table which contains " +
+    "ANSI intervals in Hive compatible format") {
+    import DayTimeIntervalType.{DAY, HOUR, MINUTE, SECOND}
+    import YearMonthIntervalType.{MONTH, YEAR}
+    withTempPath{ dir =>
+      withTable("t") {
+        val bout = new java.io.ByteArrayOutputStream
+        Console.withOut(bout) {
```

Comment:
Doesn't `LogAppender` work for you? For example:
https://github.com/apache/spark/blob/d8a4a8c629db6ae8081477e58fdbf20983b48a39/external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/v2/V2JDBCTest.scala#L170-L178

Reply:
Oh, thank you. It's better.